### ISSUE TITLE
Package downgrade warning

### ISSUE THREAD

**Issue by nulltoken** [AzureAD/azure-activedirectory-library-for-dotnet#542]
Referencing the nuget package from aspnetcore triggers the following downgrade warnings upon build

```
warn : Detected package downgrade: System.Text.RegularExpressions from 4.1.0 to 4.0.12 
warn :  MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Xml.ReaderWriter (>= 4.0.11) -> System.Text.RegularExpressions (>= 4.1.0) 
warn :  MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Text.RegularExpressions (>= 4.0.12)
warn : Detected package downgrade: System.Text.RegularExpressions from 4.1.0 to 4.0.12 
warn :  MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Runtime.Serialization.Json (>= 4.0.2) -> System.Private.DataContractSerialization (>= 4.1.1) -> System.Text.RegularExpressions (>= 4.1.0) 
warn :  MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Text.RegularExpressions (>= 4.0.12)
log  : Writing lock file to disk. Path: C:\projects\MyNeatProject\src\MyNeatProject.Web\project.lock.json
log  : C:\projects\MyNeatProject\src\MyNeatProject.Web\project.json
log  : Restore completed in 18384ms.
warn : Detected package downgrade: System.Text.RegularExpressions from 4.1.0 to 4.0.12 
warn :  MyNeatProject.Tests (>= 1.0.0) -> MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Xml.ReaderWriter (>= 4.0.11) -> System.Text.RegularExpressions (>= 4.1.0) 
warn :  MyNeatProject.Tests (>= 1.0.0) -> MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Text.RegularExpressions (>= 4.0.12)
warn : Detected package downgrade: System.Text.RegularExpressions from 4.1.0 to 4.0.12 
warn :  MyNeatProject.Tests (>= 1.0.0) -> MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Runtime.Serialization.Json (>= 4.0.2) -> System.Private.DataContractSerialization (>= 4.1.1) -> System.Text.RegularExpressions (>= 4.1.0) 
warn :  MyNeatProject.Tests (>= 1.0.0) -> MyNeatProject.Web (>= 1.0.0) -> Microsoft.IdentityModel.Clients.ActiveDirectory (>= 3.13.4) -> System.Text.RegularExpressions (>= 4.0.12)
log  : Writing lock file to disk. Path: C:\projects\MyNeatProject\test\MyNeatProject.Tests\project.lock.json
log  : C:\projects\MyNeatProject\test\MyNeatProject.Tests\project.json
log  : Restore completed in 20855ms.
```

This _seems_ to be related to the nuspec file directly referencing `System.Text.RegularExpressions` in 4.0.12.

As I can't find in the repository the script to generate the NuGet packages, I'm afraid I can't propose a Pull Request.

Not being an xplat expert, it's completely possible this is required because Platform X does things in a funny way... Any chance to get some more lights explaining what constraint led to referencing that version?

**Comment by nulltoken** [comment]
Bump? :pray:

**Comment by nulltoken** [comment]
I understand this is not a major issue. However, I'd be really grateful for some feedback.

Am I doing something wrong? Is that a known issue? Is there a "blocker" in the current build/packaging process that could prevent this dependency upgrade from happening?

Once again, I know this isn't a high priority topic. However, this kind of message adds noise (that I'd really like to get rid of) in my (shameless egotistic :wink:) build logs.

Without any feedback, I don't really know what I can do to help, or even *if* there's anything I could do.

**Comment by kpanwar** [comment]
@abhishek58g - can you investigate this please?

**Comment by kpanwar** [comment]
Fixed in 3.13.7. https://www.nuget.org/packages/Microsoft.IdentityModel.Clients.ActiveDirectory/3.13.7

**Comment by nulltoken** [comment]
@kpanwar Thanks for this. However, while upgrading, I may have bumped into another issue. Logged with #559.
